test-validator/src/lib.rs

```diff
@@ -718,14 +708,17 @@ impl TestValidatorGenesis {
         &self,
         socket_addr_space: SocketAddrSpace,
     ) -> (TestValidator, Keypair) {
+        // Using multi_thread to avoid deadlocks when client spawns tasks.
```

Comment:
Oh cool! Does this mean we don't need to do `worker-threads = 2` in tests with this change?